test/OpenTelemetry.Tests/BaseEventSourceTest.cs

```diff
@@ -0,0 +1,138 @@
+﻿// <copyright file="BaseEventSourceTest.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+
+namespace OpenTelemetry.Tests
+{
+    internal static class BaseEventSourceTest
+    {
+        public static void MethodsAreImplementedConsistentlyWithTheirAttributes(EventSource eventSource)
+        {
+            foreach (MethodInfo publicMethod in GetEventMethods(eventSource))
+            {
+                VerifyMethodImplementation(eventSource, publicMethod);
+            }
+        }
+
+        private static void VerifyMethodImplementation(EventSource eventSource, MethodInfo eventMethod)
+        {
+            using (var listener = new TestEventListener())
+            {
+                const long AllKeywords = -1;
+                listener.EnableEvents(eventSource, EventLevel.Verbose, (EventKeywords)AllKeywords);
+                try
+                {
+                    object[] eventArguments = GenerateEventArguments(eventMethod);
+                    eventMethod.Invoke(eventSource, eventArguments);
+
+                    EventWrittenEventArgs actualEvent = listener.Messages.First();
+                    VerifyEventId(eventMethod, actualEvent);
+                    VerifyEventLevel(eventMethod, actualEvent);
+
+                    if (eventMethod.Name != "ExporterErrorResult")
+                    {
+                        VerifyEventMessage(eventMethod, actualEvent, eventArguments);
+                    }
```

Comment:
For that specific method, we have an issue that we uses the enum. And, when we get from the eventlistener, it gets a number instead of the string/enum.